docs/design/jet/025-jet-job-member-pruning.md

```diff
@@ -0,0 +1,197 @@
+# Jet job member pruning
+
+|||
+|---|---|
+|Related Jira|[HZ-1605](https://hazelcast.atlassian.net/browse/HZ-1605)|
+|Document Status / Completeness|IN PROGRESS|
+|Requirement owner|Sandeep Akhouri|
+|Developer(s)|Sasha Syrotenko|
+|Quality Engineer|Isaac Sumner|
+|Technical Reviewers|TBD|
+|Version|5.4|
+
+## Background
+
+Before Hazelcast Platform 5.4, Jet job was always deployed to all data members (specifically - all not-lite members). 
+If some DAG vertex isn’t using all members, it creates no-op processors on the rest, but Jet still creates queues 
+to/from those vertices and starts the processors, even though it completes immediately and the queues are closed with a `DONE_ITEM`. 
+If some member isn’t used at all, the DAG is still deployed to it, and the master has to send it to it and wait for 
+the completion. Even though the processors are no-op, or have no data to process, it’s an unnecessary overhead, 
+which becomes noticeable in very small batch jobs.
+
+## Terminology
+
+- Member pruning - prevention of cluster members without requested data on board to be involved in job execution;
+- Processor pruning - elimination of redundant (it can work only with input) stage processor creation;
+- (IMap) Partition pruning - extract partition condition and assigning only required partition to read by `ReadMapOrCacheP`;
```

Comment:
index also can be partitioned, if it is local. Currently AFAIU we prefer global indexes, but TPC like local indexes more.

Reply:
For now we support global indices, so, no changes intended **now**

Reply:
describe partition pruning for local indexes explicitly as out-of-scope, but possible in the future

Reply:
global indexes can benefit from member pruning, added that even though the processor is currently SQL-specific